src/_pytest/outcomes.py

```diff
@@ -77,132 +73,118 @@ def __init__(
         super().__init__(msg)
 
 
-# We need a callable protocol to add attributes, for discussion see
-# https://github.com/python/mypy/issues/2087.
+class XFailed(Failed):
+    """Raised from an explicit call to pytest.xfail()."""
 
-_F = TypeVar("_F", bound=Callable[..., object])
-_ET = TypeVar("_ET", bound=type[BaseException])
 
+class _Exit:
+    Exception: type[Exit] = Exit
 
-class _WithException(Protocol[_F, _ET]):
-    Exception: _ET
-    __call__: _F
+    def __call__(self, reason: str = "", returncode: int | None = None) -> NoReturn:
+        """Exit testing process.
```

Comment:
Let's put the docstring on the class, so that sphinx picks it up correctly. You can verify that it works by checking that it shows up in the PR docs preview (currently empty): https://pytest--13445.org.readthedocs.build/en/13445/reference/reference.html#pytest-skip

Reply:
Oh this docs pipeline is neat, didn't notice that! Moved and seems like it's picked up correctly now